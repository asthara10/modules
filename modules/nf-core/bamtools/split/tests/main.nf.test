nextflow_process {

    name "Test Process BAMTOOLS_SPLIT"
    script "../main.nf"
    config "./nextflow.config"
    process "BAMTOOLS_SPLIT"

    tag "modules"
    tag "modules_nfcore"
    tag "bamtools"
    tag "bamtools/split"

    test("test_bamtools_split_single_input") {


        when {
            process {
                """
                
                input[0] = [
        [ id:'test', single_end:false ], // meta map
        file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_bam'], checkIfExists: true)
    ]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match("versions") }
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("test_bamtools_split_multiple") {
        when {
            process {
                """
                        input[0] = [
        [ id:'test', single_end:false ], // meta map
        [
            file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_bam'], checkIfExists: true),
            file(params.test_data['homo_sapiens']['illumina']['test2_paired_end_sorted_bam'], checkIfExists: true)
        ]
    ]


                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match("versions") }
                { assert snapshot(process.out).match() }
            )
        }
    }
    
}
