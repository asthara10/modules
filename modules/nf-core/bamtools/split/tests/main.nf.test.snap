{
    "versions": {
        "content": [
            [
                "versions.yml:md5,bc944a5a1a68b8b999b2ce6177c84ffc"
            ]
        ],
        "timestamp": "2023-11-24T14:17:53.709523"
    }
}